lib/modules/manager/nuget/artifacts.ts

```diff
@@ -175,15 +179,16 @@ export async function updateArtifacts({
         newLockFileContentMap[lockFileName]
       ) {
         logger.trace(`Lock file ${lockFileName} is unchanged`);
-      } else {
+      } else if (newLockFileContentMap[lockFileName]) {
         retArray.push({
           file: {
             type: 'addition',
             path: lockFileName,
-            contents: newLockFileContentMap[lockFileName],
+            contents: newLockFileContentMap[lockFileName]!,
           },
         });
       }
+      // TODO: else should we return an artifact error if new content is missing?
```

Comment:
No:

Consider the case where the user reconfigures
```
<Project>
  <PropertyGroup>
    <RestorePackagesWithLockFile>true</RestorePackagesWithLockFile>
  </PropertyGroup>
</Project>
```
to
```
<Project>
  <PropertyGroup>
    <RestorePackagesWithLockFile>false</RestorePackagesWithLockFile>
  </PropertyGroup>
</Project>
```

If the lockfiles don't match the requested version or are malformed, and `RestoreLockedMode` is configured as `true` (read: the build requires a valid lockfile), the Restore target will throw an error. There will be no need for additional logic in Renovate.

Reply:
this TODO means to return an error if there is a lockfile in repo and after running `dotnet restore` it's missing. So when somethinkg was wrong with running the dotnet executable but without error exit. Not sure if it can happen.